packages/buidler-core/src/internal/buidler-evm/provider/filter.ts

```diff
@@ -0,0 +1,130 @@
+import Bloom from "@nomiclabs/ethereumjs-vm/dist/bloom";
+import { BN, bufferToHex, toBuffer } from "ethereumjs-util";
+
+import { RpcLogOutput } from "./output";
+
+export const LATEST_BLOCK = -1;
+
+export enum Type {
+  LOGS_SUBSCRIPTION = 0,
+  PENDING_TRANSACTION_SUBSCRIPTION = 1,
+  BLOCK_SUBSCRIPTION = 2
+}
+
+export interface FilterCriteria {
+  fromBlock: number;
+  toBlock: number;
+  addresses: Buffer[];
+  topics: Array<Array<Buffer | null> | null>;
+}
+
+export interface Filter {
+  id: number;
+  type: Type;
+  criteria?: FilterCriteria;
+  deadline: Date;
+  hashes: string[];
+  logs: RpcLogOutput[];
+  subscription: boolean;
+}
+
+export function bloomFilter(
+  bloom: Bloom,
+  addresses: Buffer[],
+  topics: Array<Array<Buffer | null> | null>
+): boolean {
+  if (addresses.length > 0 && !bloom.multiCheck(addresses)) {
```

Comment:
`multickeck` only returns `true` if all the addresses are included. Don't we want something like `true` if any address is included?

Reply:
Correct :+1: